### PR TITLE
Fix issues in kms-util image

### PR DESCRIPTION
**Pull request by liu-shaojun**
This submission 
- install Java and Spark and bigdl in kms-util image so that ppml jar can be invoked to generate keys and encrypt/decrypt.
- remove pyspark-encrypt-io and spark-encrypt-io.jar from kms-util image
- update java classpath in ppml/services/kms-utils/docker/entrypoint.sh
- replace package module-init-tools with kmod, [kmod](https://launchpad.net/ubuntu/focal/amd64/kmod) replaces module-init-tools in Focal (20.04)
- replace package libprotobuf10 with libprotobuf17 since [libprotobuf10](https://packages.ubuntu.com/bionic/libprotobuf10) is only available in bionic (18.04LTS), [libprotobuf17](https://packages.ubuntu.com/focal/libprotobuf17) is for focal (20.04LTS)

Testing
- I build a new kms-util image and start a docker container. Enroll, key generation and encrypt/decrypt are tested in kms-util container.
- The new kms-util image is already pushed to 10.239.45.10/arda/intelanalytics/kms-utils:0.3.0

### CODE REVIEW

File: ppml/services/kms-utils/docker/Dockerfile [hunk 1]
```diff
@@ -1,45 +1,100 @@
...
-    cd /home && \
-    wget $SPARK_JAR_REPO_URL/spark-encrypt-io-0.2-SNAPSHOT.jar && \
-    mv spark-encrypt-io-0.2-SNAPSHOT.jar spark-encrypt-io.jar
+    git clone -b $EHSM_KMS_BRANCH_VERSION https://github.com/analytics-zoo/ehsm.git && cd ehsm && wget https://download.01.org/intel-sgx/sgx-linux/2.16/as.ld.objdump.r4.tar.gz && tar -zxf as.ld.objdump.r4.tar.gz && cp external/toolset/ubuntu18.04/* /usr/local/bin && make && cd out/ehsm-kms_enroll_app && ls ehsm-kms_enroll_app
+
 
 ADD ./entrypoint.sh /home/entrypoint.sh
-ADD ./pyspark-encrypt-io /home/pyspark-encrypt-io
+COPY --from=builder /opt/jdk /opt/jdk8
+COPY --from=builder /opt/scala-${SCALA_VERSION} /opt/scala-${SCALA_VERSION}
```
Comment (qiuxin2012) [inline ppml/services/kms-utils/docker/Dockerfile:90]: scala is just a optional dependency

Comment (liu-shaojun) [reply ppml/services/kms-utils/docker/Dockerfile]: Thanks, Scala is removed.

File: ppml/services/kms-utils/docker/Dockerfile [hunk 2]
```diff
@@ -1,45 +1,87 @@
...
+    wget https://archive.apache.org/dist/spark/spark-${SPARK_VERSION}/spark-${SPARK_VERSION}-bin-hadoop3.2.tgz && \
+    tar -zxvf spark-${SPARK_VERSION}-bin-hadoop3.2.tgz && \
+    mv spark-${SPARK_VERSION}-bin-hadoop3.2 spark-${SPARK_VERSION} && \
+    rm spark-${SPARK_VERSION}-bin-hadoop3.2.tgz && \
+    cp spark-${SPARK_VERSION}/conf/log4j.properties.template spark-${SPARK_VERSION}/conf/log4j.properties && \
+    echo $'\nlog4j.logger.io.netty=ERROR' >> spark-${SPARK_VERSION}/conf/log4j.properties
+
+# spark-encrypt-io.jar
+RUN cd /home && \
+    wget $SPARK_JAR_REPO_URL/spark-encrypt-io-0.2-SNAPSHOT.jar -O spark-encrypt-io.jar
```
Comment (glorysdj) [inline ppml/services/kms-utils/docker/Dockerfile:44]: do we still need this?

Comment (qiuxin2012) [reply ppml/services/kms-utils/docker/Dockerfile:44]: Yes, the spark-encrypt-io was migrated to BigDL PPML

Comment (liu-shaojun) [reply ppml/services/kms-utils/docker/Dockerfile]: spark-encrypt-io.jar is removed, instead, $BIGDL_HOME/jars/bigdl-ppml-spark_3.1.2-2.1.0-SNAPSHOT.jar is involved to generate keys and encrypt/decrypt.

File: ppml/services/kms-utils/docker/Dockerfile [hunk 3]
```diff
@@ -1,45 +1,101 @@
...
 ARG HTTP_PROXY_HOST
 ARG HTTP_PROXY_PORT
 ARG HTTPS_PROXY_HOST
 ARG HTTPS_PROXY_PORT
+ARG JDK_VERSION=8u192
+ARG SPARK_VERSION=3.1.2
+ARG BIGDL_VERSION=2.1.0-SNAPSHOT
+
+# stage.1 java & spark
+FROM ubuntu:18.04 AS builder
```
Comment (qiuxin2012) [inline ppml/services/kms-utils/docker/Dockerfile:10]: why change to 18.04?

Comment (liu-shaojun) [reply ppml/services/kms-utils/docker/Dockerfile]: There was package dependency error in 20.04 before, so I temporarily used 18.04, and now the package error in 20.04 has been solved, and already reverted to 20.04.

File: ppml/services/kms-utils/docker/Dockerfile [hunk 4]
```diff
@@ -1,14 +1,77 @@
...
+    ln -s /opt/jdk$JDK_VERSION /opt/jdk
+
+# spark
+RUN cd /opt && \
+    wget https://archive.apache.org/dist/spark/spark-${SPARK_VERSION}/spark-${SPARK_VERSION}-bin-hadoop3.2.tgz && \
+    tar -zxvf spark-${SPARK_VERSION}-bin-hadoop3.2.tgz && \
+    mv spark-${SPARK_VERSION}-bin-hadoop3.2 spark-${SPARK_VERSION} && \
+    rm spark-${SPARK_VERSION}-bin-hadoop3.2.tgz && \
+    cp spark-${SPARK_VERSION}/conf/log4j.properties.template spark-${SPARK_VERSION}/conf/log4j.properties && \
+    echo $'\nlog4j.logger.io.netty=ERROR' >> spark-${SPARK_VERSION}/conf/log4j.properties
```
Comment (glorysdj) [inline ppml/services/kms-utils/docker/Dockerfile:41]: should rm log4j1 and use log4j2
please refer to https://github.com/intel-analytics/BigDL/blob/main/docker/bigdl-k8s/Dockerfile#L37

Comment (liu-shaojun) [reply ppml/services/kms-utils/docker/Dockerfile]: updated

**Review by glorysdj — APPROVED** [review]
LGTM

**Pull request merged**